sql/catalyst/src/main/scala/org/apache/spark/sql/types/FloatType.scala

```diff
@@ -78,8 +78,12 @@ case object FloatType extends FloatType {
   }
 
   trait FloatAsIfIntegral extends FloatIsConflicted with Integral[Float] {
-    def quot(x: Float, y: Float): Float = (BigDecimal(x) quot BigDecimal(y)).floatValue
-    def rem(x: Float, y: Float): Float = (BigDecimal(x) remainder BigDecimal(y)).floatValue
+    def quot(x: Float, y: Float): Float = {
```

Comment:
```
Warning:Warning:line (81)method apply in object BigDecimal is deprecated (since 2.11.0): The default conversion from Float may not do what you want. Use BigDecimal.decimal for a String representation, or explicitly convert the Float with .toDouble.
```

Reply:
This code I copied from Scala 2.12 very recently. I think this is probably OK, but I don't know the implications of it.

Reply:
If we are not sure about the implications, shall we revert this part and merge this PR, @MaxGekk ?

Reply:
I think we can leave it, just because this is the implementation:
```
  def apply(x: Float): BigDecimal = apply(x.toDouble)
```
I am not sure why it's deprecated, or what the 'default conversion' issue is. I understand the issue with double/float not being exact representations, but then why is `apply(Double)` not deprecated?

Reply:
> I am not sure why it's deprecated, or what the 'default conversion' issue is.

The code was removed by https://github.com/scala/scala/pull/6468/files#diff-a1c7fbbd4efafafc57671a3ac783aaeaL143-L149 due to troubles with Double/Float ranges.

Reply:
Yep it's gone in 2.13, which is why I just inlined it back here for 2.13 + 2.12 cross compilation. I get why scala is punting a lot on dealing with floating-point ordering. But to this point, I don't see why double would be different. DoubleAsIfIntegral went away too. Anyway, at least here, this change seems OK as it does not actually change behavior at all.

Reply:
Got it. I'll leave this PR to @srowen .